msgraph/core/middleware/authorization.py

```diff
@@ -2,35 +2,39 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 # ------------------------------------
+from typing import TypeVar
+
+import httpx
+from kiota_abstractions.authentication import AccessTokenProvider
+from kiota_http.middleware.middleware import BaseMiddleware
+
 from .._enums import FeatureUsageFlag
-from .abc_token_credential import TokenCredential
-from .middleware import BaseMiddleware
+from .middleware import GraphRequest
+
+
+class GraphAuthorizationHandler(BaseMiddleware):
```

Comment:
Just to confirm, isn't this already being done by the `RequestAdapter` by the relevant `IAuthenticationProvider` making this redundant?

Reply:
The http package will be doing a double work according to https://github.com/microsoft/kiota-http-python/blob/cc640f885aa32c097b9eb0189189b50d3dbba979/kiota_http/httpx_request_adapter.py#L315

Reply:
Users who want to use the core directly without a service library will need a simple way to authenticate, outside of the RequestAdapter

Reply:
> The http package will be doing a double work according to https://github.com/microsoft/kiota-http-python/blob/cc640f885aa32c097b9eb0189189b50d3dbba979/kiota_http/httpx_request_adapter.py#L315

Not really the case. `GraphRequestAdapter` reuses the work done in RequestAdapter and does not utilize the Authorization middleware. Only the `GraphClient` will use it, as it is designed to be used for making direct http requests

Reply:
Will resolve this for now. As this is sort of covered by the discussion at https://github.com/microsoftgraph/msgraph-sdk-python-core/pull/141#discussion_r1009374364.
If the `RequestAdapter` can be used to send the request using `RequestInformation` instances, we won't need this.